kedro-datasets/kedro_datasets/pandas/csv_dataset.py

```diff
@@ -188,3 +188,11 @@ def _invalidate_cache(self) -> None:
         """Invalidate underlying filesystem caches."""
         filepath = get_filepath_str(self._filepath, self._protocol)
         self._fs.invalidate_cache(filepath)
+
+    def _preview(self, nrows) -> Dict:
+        # Create a copy so it doesn't contaminate the original dataset
+        dataset_copy = self._copy()
+        dataset_copy._load_args["nrows"] = nrows
+        data = dataset_copy.load()
+
+        return data.to_dict()
```

Comment:
Wait, I just realised the same options are also available in the `orient` argument of `to_dict`: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_dict.html. So forget about using `to_json`, but think about whether any of those different `orient` values might be useful.

Reply:
e.g. I think `split` or `tight` will more directly correspond to how `<tr>` and `<td>` work, so it will make the mapping in your `PreviewTable` simpler.